x/ccv/provider/keeper/proposal.go

```diff
@@ -544,3 +562,19 @@ func (k Keeper) CloseChannel(ctx sdk.Context, channelID string) {
 		}
 	}
 }
+
+// CreateConsumerClientInCachedCtx creates a consumer client
+// from a given consumer addition proposal in a cached context
+func (k Keeper) CreateConsumerClientInCachedCtx(ctx sdk.Context, p types.ConsumerAdditionProposal) (cc sdk.Context, writeCache func(), err error) {
```

Comment:
Valid point but I prefer the `...CachedCtx` naming scheme bc it's more generic so that it can be used in `BeginBlockInit` and `HandleConsumerAdditionProposal`.